src/open-websocket.js

```diff
@@ -0,0 +1,14 @@
+import Html5WebSocket from 'html5-websocket'
+import ReconnectingWebSocket from 'reconnecting-websocket'
+import WebSocket from 'ws'
+
+export default wsAddressBuilder => {
+  return new ReconnectingWebSocket(wsAddressBuilder, undefined, {
+    connectionTimeout: 4000,
+    constructor: typeof window !== 'undefined' ? WebSocket : Html5WebSocket,
```

Comment:
Wondering about this conditional, would it use `ws` under a browser environment and HtmlWebSocket under node? Shouldn't it be the opposite?

Reply:
When `window` is defined, then it uses the standard [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) interface. Otherwise it uses [Html5WebSocket](https://github.com/pladaria/html5-websocket), which is a thin layer for the `ws` npm package.

Reply:
Then we probably want to remove the `ws` import (and explicit dependency) which is going to override `WebSocket` in this context no?